src/test/scala/cognite/spark/v1/FilesRelationTest.scala

```diff
@@ -39,74 +55,300 @@ class FilesRelationTest extends FlatSpec with Matchers with SparkTest {
+                |null as id,
+                |'$source' as source,
+                |'externalId-$source' as externalId,
+                |null as mimeType,
+                |null as metadata,
+                |null as assetIds,
+                |null as datasetId,
+                |null as sourceCreatedTime,
+                |null as sourceModifiedTime,
+                |null as securityCategories,
+                |null as uploaded,
+                |null as createdTime,
+                |null as lastUpdatedTime,
+                |null as uploadedTime,
+                |null as uploadUrl
      """.stripMargin)
-      .select(df.columns.map(col): _*)
-      .write
-      .insertInto("filesSource")
+        .select(sourceDf.columns.map(col): _*)
+        .write
+        .insertInto("destinationFilesUpdate")
 
-    val dfWithTestDirectory = retryWhile[List[File]](
-      writeClient.files.filter(FilesFilter(assetIds = Some(Seq(1739842716040355L)))).compile.toList,
-      files => files.length < 10)
-    assert(dfWithTestDirectory.length == 10)
+      val rows = retryWhile[Array[Row]](
+        spark.sql(s"""select * from destinationFilesUpdate where source = "$source"""").collect,
+        rows => rows.length < 1)
+      assert(rows.length == 1)
 
-    spark
-      .sql(s"""
-            |select id,
-            |name,
-            |source,
-            |id as externalId,
-            |mimetype,
-            |metadata,
-            |Array(176631399357975) as assetIds,
-            |uploaded,
-            |uploadedTime,
-            |createdTime,
-            |lastUpdatedTime,
-            |uploadUrl,
-            |dataSetId
-            |from filesSource
-            |where source = '$originalSource'
+      val filesCreated = getNumberOfRowsCreated(metricsPrefix, "files")
+      assert(filesCreated == 1)
+
+      val id = writeClient.files.retrieveByExternalId(s"externalId-$source").id
+
+      //Update using id
+      spark
+        .sql(s"""
+                |select ${id.toString} as id,
+                |'$source' as source,
+                |'updatedById-externalId-$source' as externalId
      """.stripMargin)
-      .select(df.columns.map(col): _*)
-      .write
-      .insertInto("filesSource")
+        .write
+        .format("cognite.spark.v1")
+        .option("apiKey", writeApiKey)
+        .option("type", "files")
+        .option("onconflict", "update")
+        .option("collectMetrics", "true")
+        .option("metricsPrefix", metricsPrefix)
+        .save()
 
-    val dfWithUpdatedSource = retryWhile[List[File]](
-      writeClient.files.filter(FilesFilter(assetIds = Some(Seq(176631399357975L)))).compile.toList,
-      files => files.length < 10)
-    assert(dfWithUpdatedSource.length == 10)
+      val filesCreated2 = getNumberOfRowsCreated(metricsPrefix, "files")
+      assert(filesCreated2 == 1)
+      var updatedFiles = getNumberOfRowsUpdated(metricsPrefix, "files")
+      assert(updatedFiles == 1)
```

Comment:
Don't use `var`, but if only checking the value once and then not using it again, you can do `assert(getNumberOfRowsUpdated(metricsPrefix, "files") == 1)`

Reply:
I was checking two updates (one by id and other by externalId).
will just write assert(getNumberOfRowsUpdated(metricsPrefix, "files") == 1).